plugins/org.komodo.shell/src/org/komodo/shell/DefaultKomodoShell.java

```diff
@@ -0,0 +1,293 @@
+        if (locale_str != null) {
+            String lang = null;
+            String region = null;
+            String[] lsplit = locale_str.split("_"); //$NON-NLS-1$
+            if (lsplit.length > 0) {
+                lang = lsplit[0];
+            }
+            if (lsplit.length > 1) {
+                region = lsplit[1];
+            }
+            if (lang != null && region != null) {
+                Locale.setDefault(new Locale(lang, region));
+            } else if (lang != null) {
+                Locale.setDefault(new Locale(lang));
+            }
+        }
+
+        final KomodoShellParent parent = new KomodoShellParent() {
+
+            @Override
+            public void exit() {
+                System.exit(0);
+            }
+        };
+
+        final DefaultKomodoShell shell = new DefaultKomodoShell(parent, KEngine.getInstance(), System.in, System.out);
+        Thread shutdownHook = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                shell.shutdown();
+            }
+        });
+        Runtime.getRuntime().addShutdownHook(shutdownHook);
+        try {
+            shell.run(args);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private WorkspaceStatus wsStatus;
+    private ShellCommandFactory factory;
+    private ShellCommandReader reader;
+    private boolean shutdown = false;
+
+    private final KomodoShellParent parent;
+    private final KEngine kEngine;
+    private final InputStream inStream;
+    private final PrintStream outStream;
+
+    /**
+     * Constructor.
+     * @param parent the UI parent of this shell
+     * @param kEngine the komodo engine for this shell
+     * @param inStream input stream
+     * @param outStream output stream
+     */
+    public DefaultKomodoShell(KomodoShellParent parent, KEngine kEngine, InputStream inStream, PrintStream outStream) {
+        this.parent = parent;
+        this.kEngine = kEngine;
+        this.inStream = inStream;
+        this.outStream = outStream;
+
+        StringBuffer sb = new StringBuffer();
+        for (int i = 0; i < CompletionConstants.MESSAGE_INDENT; i++) {
+            sb.append(StringConstants.SPACE);
+        }
+        msgIndentStr = sb.toString();
+    }
+
+    @Override
+    public KEngine getEngine() {
+        return kEngine;
+    }
+
+    @Override
+    public InputStream getInputStream() {
+        return inStream;
+    }
+
+    @Override
+    public PrintStream getOutputStream() {
+        return outStream;
+    }
+
+    @Override
+    public void exit() {
+        shutdown();
+        parent.exit();
+    }
+
+    /**
+     * Runs the shell.
+     * @param args the arguments
+     * @throws Exception the exception
+     */
+    public void run(String[] args) throws Exception {
+        // This will block and await the start of both the engine and its default repository
+        startKEngine();
```

Comment:
Is there any point in executing the rest of this method if the engine doesn't start? Since we are blocking here, I'm wondering if `startKEngine()` should return a `boolean` and then the rest of this `run` method could be avoided.